library/core/src/slice/cmp.rs

```diff
@@ -60,7 +60,24 @@ where
             return false;
         }
 
-        self.iter().zip(other.iter()).all(|(x, y)| x == y)
+        let mut i = self.len();
+        let mut ptr_self = self.as_ptr();
+        let mut ptr_other = other.as_ptr();
+        // SAFETY:
+        // This is sound because:
+        // - self.len == other.len
+        // - self.len <= isize::MAX
```

Comment:
That isn't true for ZSTs. Though the result still happens to work because bumping zst pointers does nothing

Reply:
Thanks for the review!